Improve SharpHound permissions documentation structure and cross-referencing (#77)
SpecterOps/bloodhound-docs: docs/collect-data/enterprise-collection/least-privileged-collection.mdx

You can delegate permissions to a group for [read access to the "Deleted Objects" container](https://learn.microsoft.com/en-us/troubleshoot/windows-server/identity/non-administrators-view-deleted-object-container), and then add the SharpHound collector service account to that group.

The Deleted Objects container locations are:
* Domain NC Deleted Objects DistinguishedName (DN): `CN=Deleted Objects,<Domain DN>`
* Configuration NC Deleted Objects DistinguishedName (DN): `CN=Deleted Objects,CN=Configuration,<Forest root domain DN>`

<Note>SharpHound can read these containers even if the AD Recycle Bin feature is not enabled.</Note>

## Local Group Membership

SharpHound collects local group membership via [Remote SAM Enumeration](https://learn.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/network-access-restrict-clients-allowed-to-make-remote-sam-calls).

> By default, on currently supported Windows operating systems, only `Administrators` on the device(s) being collected have this right on Windows clients and member servers.  For compatibility purposes, `Everyone` is granted this right on domain controllers by default.

Reviewer: I'm not sure there's anything wrong with what its saying.  It's a nitpick comment being nitpicky.  In trying to keep things brief I am referenced the `Everyone` special identity group, which includes Authenticated Users and Guests in current OSes.  Not the collective "Everyone" as in "Anyone".  Everyone only included Anonymous access in Server 2000.  

DCs do ignore the GPO setting as it states.  It's wording the same thing differently and with more words.

### Deleted Objects Container (Optional)

SharpHound can read the contents of the Deleted Objects container (also known as the AD Recycle Bin). Collecting deleted objects affects data retention behavior in BloodHound Enterprise—see [Active Directory Recycle Bin](/collect-data/enterprise-collection/data-retention#active-directory-recycle-bin) for details on how this impacts retention periods.

To grant the SharpHound service account [read access to the "Deleted Objects" container](https://learn.microsoft.com/en-us/troubleshoot/windows-server/identity/non-administrators-view-deleted-object-container), permissions may be delegated to a group and the SharpHound collector service account made a member of that group.

The Deleted Objects container locations are:
* Domain NC Deleted Objects DistinguishedName (DN): `CN=Deleted Objects,<Domain DN>`
* Configuration NC Deleted Objects DistinguishedName (DN): `CN=Deleted Objects,CN=Configuration,<Forest root domain DN>`

> <Note>It's not required to have the AD recycle bin feature enabled for SharpHound to read these containers.</Note>

Reviewer 2: ```suggestion
<Note>SharpHound can read these containers even if the AD Recycle Bin feature is not enabled.</Note>
```

SharpHound collects registry data for both certificate services and NTLM relay edges. The certificate services registry paths are on certificate authorities and domain controllers. NTLM relay paths are located on all Windows hosts.

By default, only Administrators may read the registry remotely. There are two methods to delegate remote registry access for least-privileged collection:

> 1. **AllowedPaths / AllowedExactPaths exceptions**: Create exceptions using the GPO setting [Network access: Remotely accessible registry paths](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-10/security/threat-protection/security-policy-settings/network-access-remotely-accessible-registry-paths) to allow Authenticated Users to connect to the Remote Registry named pipe at specific registry paths. Effective access is still governed by permissions on individual registry keys.

Reviewer 2: ```suggestion
1. **AllowedPaths / AllowedExactPaths exceptions**: Create exceptions using the GPO setting [Network access: Remotely accessible registry paths](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-10/security/threat-protection/security-policy-settings/network-access-remotely-accessible-registry-paths) to allow Authenticated Users to connect to the Remote Registry named pipe at specific registry paths. Permissions on individual registry keys still govern effective access.
```

> 1. **AllowedPaths / AllowedExactPaths exceptions**: Create exceptions using the GPO setting [Network access: Remotely accessible registry paths](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-10/security/threat-protection/security-policy-settings/network-access-remotely-accessible-registry-paths) to allow Authenticated Users to connect to the Remote Registry named pipe at specific registry paths. Effective access is still governed by permissions on individual registry keys.

Reviewer 3: Should be "individual registry key" (no s)

2. **Modify Remote Registry named pipe security descriptor**: Alternatively, modify the default security descriptor on the Remote Registry named pipe to grant explicit security principals read permissions to the entire registry. Effective access is still governed via permissions configured on individual registry keys.

<Frame>
  <img src="/assets/least-priv-gpo-winreg.png" alt=""/>
</Frame>

### CA Registry

> When the AD CS role is installed in Windows, the `HKLM\SYSTEM\CurrentControlSet\Services\CertSvc` registry path is automatically added to `HKLM\SYSTEM\CurrentControlSet\Control\SecurePipeServers\winreg\AllowedPaths`. This creates a remote registry exception that allows Authenticated Users to query any keys and subkeys in this path, as long as they also are granted rights to read the key. Therefore, CA registry data is accessible to Authenticated Users by default when AD CS is installed.

Reviewer 3: I would change for `Authenticated Users` to make it obvious it's a group.

### DC Registry

> For domain controllers, you can create exceptions by adding the required DC registry paths to `HKLM\SYSTEM\CurrentControlSet\Control\SecurePipeServers\winreg\AllowedExactPaths` using the GPO setting mentioned above. This will create an exception to the Remote Registry named pipe on the DC allowing Authenticated Users to read those exact key paths, as long as they also are granted permissions on the registry key DACL.

Reviewer 3: Same thing here, use backticks around `Authenticated Users`. 
The last part of the sentence is hard to understand. 
Can you rephrase "llowing Authenticated Users to read those exact key paths, as long as they also are granted permissions on the registry key DACL."?

### NTLM Relay Registry Paths

> The registry paths for NTLM relay edges are located on all Windows hosts and are only accessible remotely by Administrators by default. Group Policy can be utilized to create AllowedExactPaths exceptions for these specific paths using the same method described above.

Reviewer 2: ```suggestion
Registry paths for NTLM relay edges exist on all Windows hosts. By default, only Administrators can access these paths remotely. To enable least-privileged collection, use Group Policy to add these specific paths to AllowedExactPaths. This grants Authenticated Users remote read access, following the same method described above.
```
